core-ui/src/shared/ResourceForm/ResourceForm.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import LuigiClient from '@luigi-project/client';
+import { useNotification } from 'react-shared';
+import { useTranslation } from 'react-i18next';
+import * as jp from 'jsonpath';
+import { ModeSelector } from './components/ModeSelector';
+import { Editor } from './components/Editor';
+import { Presets } from './components/Presets';
+import * as FormComponents from './components/FormComponents';
+import './ResourceForm.scss';
+
+ResourceForm.Label = FormComponents.Label;
+ResourceForm.Input = FormComponents.Input;
+ResourceForm.CollapsibleSection = FormComponents.CollapsibleSection;
+ResourceForm.FormField = FormComponents.FormField;
+ResourceForm.TextArrayInput = FormComponents.TextArrayInput;
+ResourceForm.K8sNameField = FormComponents.K8sNameField;
+ResourceForm.KeyValueField = FormComponents.KeyValueField;
+
+export function ResourceForm({
+  pluralKind, // used for the request path
+  nameSingular,
+  resource,
+  setResource,
+  onCreate,
+  onChange,
+  formElementRef,
+  children,
+  renderEditor,
+  createFn,
+  presets,
+  onPresetSelected,
+}) {
+  const notification = useNotification();
+  const { t } = useTranslation();
+  const [mode, setMode] = React.useState(ModeSelector.MODE_SIMPLE);
+
+  if (!onCreate) {
+    onCreate = async () => {
```

Comment:
not critical, but I'm not sure about including this function in here - I think it might be a better idea to separate it to a completely standalone `resourceCreate` so it's more reusable outside of this form (for example in a wizard).

And perhaps even generalize it a bit more (for example add an option whether to redirect or not) so it can be used in Deployments as well - as two subsequent calls to it (one for deployments, one for services).

Might not be worth it, though. So it's not a requirement :)

Reply:
That's a good point - I've created a `useResourceCreate` hook with option to override the behaviour after successful request.

And used it for Deployments, too.

It was worth it.